ports/superdungeonmuncher/Super Dungeon Muncher.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+XDG_DATA_HOME=${XDG_DATA_HOME:-$HOME/.local/share}
+
+if [ -d "/opt/system/Tools/PortMaster/" ]; then
+  controlfolder="/opt/system/Tools/PortMaster"
+elif [ -d "/opt/tools/PortMaster/" ]; then
+  controlfolder="/opt/tools/PortMaster"
+elif [ -d "$XDG_DATA_HOME/PortMaster/" ]; then
+  controlfolder="$XDG_DATA_HOME/PortMaster"
+else
+  controlfolder="/roms/ports/PortMaster"
+fi
+
+source $controlfolder/control.txt
+source $controlfolder/device_info.txt
+
+get_controls
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+
+$ESUDO chmod 666 /dev/tty0
+
+GAMEDIR="/$directory/ports/superdungeonmuncher"
+
+# We log the execution of the script into log.txt
+exec > >(tee "$GAMEDIR/log.txt") 2>&1
+
+export LD_LIBRARY_PATH="/usr/lib:$GAMEDIR/lib:$LD_LIBRARY_PATH"
+
+cd $GAMEDIR
+
+# rename game files for gmloadernext
+if [ -f "data.win" ]; then
+    mv data.win game.droid
+fi
+
+# Make sure uinput is accessible so we can make use of the gptokeyb controls
+$ESUDO chmod 666 /dev/uinput
+
+$GPTOKEYB "gmloadernext" -c ./control.gptk &
```

Comment:
If you make a file `control.gptk` and put it in your `superdungeonmuncher` folder it will automatically work. You can borrow this from `Undertale Yellow`; in a gptk file, a `\"` means "defer to SDL". Otherwise, you can assign keyboard keys to buttons.

Reply:
thanks! learned something :)